EnhancedMediaPlayer/MediaPlayerView.swift

```diff
@@ -15,107 +14,52 @@ public struct MediaPlayerView: View {
+            ))
+
+            if viewModel.areControlsVisible {
+                MediaPlayerControlsView(viewModel: .init(
+                    screenWidth: $screenWidth,
+                    playerState: $viewModel.playerState,
+                    seekFactor: viewModel.seekFactor,
+                    onTapAction: viewModel.onTapAction
+                ))
             }
         }
         .readFrame { size in
             screenWidth = size.width
         }
     }
-    
-    @ViewBuilder private func renderOverlay() -> some View {
-        Rectangle()
-            .fill(.black)
-            .opacity(Constants.overlayOpacity)
-            .frame(maxWidth: .infinity, maxHeight: .infinity)
-    }
-
-    @ViewBuilder private func renderActionButtons() -> some View {
-        HStack(spacing: .zero) {
-            renderRewind()
-                .frame(maxWidth: .infinity, alignment: .leading)
-            if viewModel.playerState == .finished {
-                renderReplay()
-            } else {
-                renderPlayPause()
-            }
-            renderForward()
-                .frame(maxWidth: .infinity, alignment: .trailing)
-        }
-        .frame(maxWidth: .infinity)
-    }
-
-    @ViewBuilder private func renderPlayPause() -> some View {
-        renderButton(control: viewModel.playerState == .playing ? .pause : .play)
-    }
-    
-    @ViewBuilder private func renderReplay() -> some View {
-        renderButton(control: .replay)
-    }
-    
-    @ViewBuilder private func renderRewind() -> some View {
-        renderTimeElapsedWrapper {
-            renderButton(control: .rewind)
-        }
-    }
-    
-    @ViewBuilder private func renderForward() -> some View {
-        renderTimeElapsedWrapper {
-            renderButton(control: .forward)
-        }
-    }
-    
-    @ViewBuilder private func renderButton(control: Control) -> some View {
-        control.systemImage
-            .resizable()
-            .frame(width: Constants.iconSize.width, height: Constants.iconSize.height)
-            .foregroundColor(.white)
-            .wrapInButton(action: { viewModel.onTapAction?(control) })
-    }
-    
-    @ViewBuilder private func renderTimeElapsedWrapper<Content: View>(
-        @ViewBuilder content: @escaping () -> Content
-    ) -> some View {
-        ZStack {
-            content()
-            Text("10") // TODO: update with amount of rewind / forward
-                .foregroundColor(.white)
-                .font(.system(size: Constants.timeFontSize))
-        }
-    }
 }
 
 extension MediaPlayerView {
     public class ViewModel: ObservableObject {
+        @Published var isSeekByTapEnabled: Bool = true
         @Published var playerState: PlayerState = .loading
+        @Published var seekState: MediaPlayerTappableView.SeekState = .none
+        @Published var areControlsVisible: Bool = false
         @Published var player: AVPlayer
+        @Published var visibleSeek: MediaPlayerTappableView.Seek?
```

Comment:
Ditto about this `visibleSeek` not needing to exist, it should be a computed variable in `SeekState`